packages/core/src/storages/request_queue.ts

```diff
@@ -200,10 +200,64 @@ export class RequestQueue {
     constructor(options: RequestQueueOptions) {
         this.id = options.id;
         this.name = options.name;
-        this.client = options.client.requestQueue(this.id, {
+        const client = options.client.requestQueue(this.id, {
             clientKey: this.clientKey,
             timeoutSecs: this.timeoutSecs,
         }) as RequestQueueClient;
+
+        // dirty quick hack to serialize the Request.internalVariables into Request.userData.
+        // needs to be solved somehow else.
+        // TODO: requestQueue tests need to be fixed. they are getting confused with the internalVariables and userData shenanigans.
+        this.client = new Proxy(client, {
+            get(target, prop: keyof RequestQueueClient) {
```

Comment:
Can't we just override the [`toJSON`](https://javascript.info/json#custom-tojson) function of `Request` to achieve this? Or add a `request.serialize()` method and call it everywhere before passing the Request to client?

If it can be done (that largely depends on how we serialize in `apify-client`), I would prefer the `toJSON` way, because it would work even when the user decides to do something with the request manually.

Reply:
I like the `toJSON` solution, combines well with the non-enumerable property definition suggested above. Although as you say, it expects `apify-client` to JSONify the data in the first place. Again, you guys are the overlords, if this is OK, it's probably the cleanest solution idea so far :)